pkgs/by-name/ko/koboldcpp/package.nix

```
@@ -0,0 +1,115 @@
+{ lib
+, fetchFromGitHub
+, stdenv
+, autoPatchelfHook
+, makeWrapper
+, python311Packages
+, tk
+
+, koboldLiteSupport ? true
+
+, config
+, cudaPackages ? { }
+
+, openblasSupport ? !stdenv.isDarwin
+, openblas
+
+, cublasSupport ? config.cudaSupport
+
+, clblastSupport ? stdenv.isLinux
+, clblast
+, ocl-icd
+
+, vulkanSupport ? true
+, vulkan-loader
+
+, metalSupport ? stdenv.isDarwin && stdenv.isAarch64
+
+, march ? ""
+, mtune ? ""
+}:
+
+let
+  inherit (lib) optionals optionalString;
+
+  pythonInputs = with python311Packages; [
+    tkinter
+    customtkinter
+    packaging
+    tk
+  ];
+
+  cudaBuildInputs = with cudaPackages; [
+      libcublas
+  ];
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "koboldcpp";
+  version = "1.65";
+
+  src = fetchFromGitHub {
+    owner = "LostRuins";
+    repo = "koboldcpp";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-GYvU+0bhskCgSXAoeSy/2wWIF5FrilLVdXitFvuUAdQ=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    python311Packages.wrapPython
+  ];
+
+  buildInputs = pythonInputs ++ optionals openblasSupport [ openblas ]
+  ++ optionals cublasSupport [ cudaPackages.libcublas ]
+  ++ optionals clblastSupport [ clblast ocl-icd ]
+  ++ optionals vulkanSupport [ vulkan-loader ];
+
+  pythonPath = pythonInputs;
+
+  NIX_CFLAGS_COMPILE = optionalString (march != "") "-march=${march}"
+  + optionalString (mtune != "") "-mtune=${mtune}";
+
+  makeFlags = optionals openblasSupport [ "LLAMA_OPENBLAS=1" ]
+  ++ optionals cublasSupport [ "LLAMA_CUBLAS=1" ]
+  ++ optionals clblastSupport [ "LLAMA_CLBLAST=1" ]
+  ++ optionals vulkanSupport [ "LLAMA_VULKAN=1" ]
+  ++ optionals metalSupport [ "LLAMA_METAL=1" ];
+
+  enableParallelBuilding = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p "$out/bin"
+
+    install -Dm755 koboldcpp.py "$out/bin/koboldcpp.unwrapped"
+    cp *.so "$out/bin"
+    cp *.embd "$out/bin"
+
+    if [ ${lib.boolToString koboldLiteSupport} == "false" ]; then
+      echo "koboldLiteSupport disabled, removing appropriate filew."
+      rm "$out/bin/kcpp_docs.embd"
+      rm "$out/bin/klite.embd"
+    fi
```

```suggestion
    ${optionalString (!koboldLiteSupport) ''
      rm "$out/bin/kcpp_docs.embd"
      rm "$out/bin/klite.embd"
    ''}
```
I think this is a better approach. Also, since the option is enabled by default, I don't think it's worth leaving a comment with `echo`